lib/grant.js

```diff
@@ -449,7 +452,24 @@ function saveRefreshToken (done) {
 }
 
 /**
- * Create an access token and save it with the model
+ * Pass the scope string to the model for saving
+ *
+ * @param  {Function} done
+ * @this   OAuth
+ */
+function saveScope (done) {
+  var scope = this.scope || this.req.body.scope;
+
+  var self = this;
+  this.model.saveScope(scope, self.accessToken, function (err, acceptedScope) {
```

Comment:
The *saveScope* function will be called on the model even if the user did not send a scope. In fact, *scope* here will be `undefined`.

I suggest following the pattern set by the `saveRefreshToken` function (and even your `checkScope`):

```js
if (!scope) return done();
```